source/s3_client.c

```diff
@@ -1088,8 +1088,13 @@ struct aws_s3_meta_request *aws_s3_client_make_meta_request(
             aws_error_str(aws_last_error()));
 
         meta_request = aws_s3_meta_request_release(meta_request);
-    } else {
+    } else if (meta_request != NULL) {
```

Comment:
I don't think it's possible for meta_request to be NULL in this `else` block?

If it were possible, we wouldn't end up logging anything about whether the meta-request creation succeeded or failed

```suggestion
    } else {
```